src/main/java/menion/android/whereyougo/gui/extension/activity/CustomActivity.java

```diff
@@ -40,10 +46,16 @@
 import menion.android.whereyougo.utils.NotificationService;
 
 public class CustomActivity extends FragmentActivity {
+    private static final String TAG = CustomActivity.class.getSimpleName();
+
+    protected void customOnCreate(Activity activity) {
+        Logger.v(TAG, "customOnCreate(), id:" + activity.hashCode());
+
+        // Set language
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+        String lang = sharedPreferences.getString(getString(R.string.pref_KEY_S_LANGUAGE), "");
+        setLocale(this, lang);
```

Comment:
Why do we need the language management in different classes (here and in XmlSettingsActivity)?

Reply:
Yes because the settings are not inheriting from the `CustomActivity` anymore but instead are inheriting from `AppCompatActivity`. Otherwise we need to change the implementation of the `CustomActivity` and this would have drastic effects for the whole application. That' why I duplicated this part. If you desire I can try out if it is possible to use the `AppCompatActivity` as a base class for the `CustomActivity`.